### ISSUE TITLE
❤️ Looking for new Project Owners 🤓

### ISSUE THREAD

**Issue by freak4pc** [RxSwiftCommunity/contributors#67]
# Project Spearhead 𐃉
## Looking for Project Stewards (Owners)

**tl;dr**: We're looking for people to become owners/managers who lead specific RxSwift Community projects under full ownership/autonomy, while deciding the future and spirit of each individual project and keeping the community spirit of it in general.

------

### State of the Union

The RxSwift Community organization has exploded with dozens of awesome and useful projects over the past years. Up until today, the process for getting your project into RxSwiftCommunity was mostly set and straightforward. But keeping the projects matintained in the long-term proved to be difficult: 

1. Some people just moved away from their projects or stopped doing open-source all together
2. Projects kept being maintained by kind and random acts, but without any guiding force. People usually tag me in every single repo to merge requests and cut releases which is sub-optimal and takes a lot of effort on my end.
3. When there are questions specific to a project, there isn't really an obvious owner to talk to about the future and direction of the project.
4. The RxSwiftCommunity organization incluldes many pojects that haven't been maintained for well over a year. Some of them aren't even on RxSwift 5 / Swift 5.

For these reasons and many more, a discussion that first started a few years ago with @ashfurrow came back to mind. 

This idea will be the guiding force I aim to take RxSwift Community's future into. We want to have an **owner/point person for every project**. What does this mean? 

* Projects are still community-based and maintained by the community in general.
* But, each project would have a single owner (or more) to guide its future and direction. This person would be mostly autonomous and have basically complete ownership over a specific project, cutting releases, deciding what to merge into a specific project, etc.

We already have some projects that have these "people" but we never officially stamped it. 

This is a big change and effort but I feel confident this will increase the maintenance level of most projecs, and also shine light on the increidble people in this community who work super hard on maintaining its projects.

> **Note**: As for the last point, I personally plan to look at very old and unmantained projects, and possibly remove them from the organizartion if no owner is found for them. This is for the greater good in this case, since we all want to have only well-maintained projecs and not just "dead" ones.

### Sounds awesome, I'm In! 

Take a look at the table below, or any other RxSwiftCommunity repo which I didn't mention here. If it currently doesn't have an owner and you are interested in taking ownership over it, please e-mail me at [shai@rxswift.org](mailto:shai@rxswift.org) or reach out on Twitter (@freak4pc) and we'll discuss to get you started. 

Each project will get their own e-mail forwarder (i.e. rxalamofire@rxswift.org) so they can cut releases as an owner for CocoaPods, etc. 

### Current owners

If you want to steward an unowned project in this list, please reach out to me on shai@rxswift.org or on Twitter (@freak4pc).

| Project        | Owner(s)           | Original Author(s)  |
|----------------|--------------------|---------------------|
| RxSwiftExt     | @freak4pc          | @fpillet, @icanzilb |
| RxDataSources  | @freak4pc, @kzaher | @kzaher             |
| RxCoreLocation | @bobgodwinx        | @bobgodwinx         |
| RxKingfisher   | @freak4pc          | @freak4pc           |
| RxKeyboard     | @jdisho	          | @devxoul            |
| RxGRDB         | @groue             | @groue              |
| RxFlow         | @twittemb          | @twittemb           |
| RxRealm        | None.              | @icanzilb           |
| RxAlamofire    | @rynecheow         | @bontoJR            |
| Action         | @bobgodwinx 		  | @ashfurrow          |
| RxGesture      | @jegnux            | @jegnux             |
| RxMKMapView    | @minuscorp         | @sger               |
| RxMarbles      | @yury              | @yury               |
| RxStarscream   | @fassko            | @GuyKahlon          |
| RxFirebase     | @engali94          | @arnauddorgans      |
| RxAnimated     | @jdisho, @fassko   | @icanzilb           |
| RxReachability | @JoeMatt           | @ivanbruel          |
| RxWebKit       | @fabianmuecke, @mRs- | @daichiro           |
| RxCoreMotion   | @rizwan95          | @carlosypunto       |
| RxCoreData	 | **None**			  | @scotteg 			|
| RxIGListKit    | @Bruce-pac		  | @Bruce-pac 			|

If I missed any projects, please let me know :]

**Comment by bobgodwinx** [comment]
@freak4pc You can assign `Actions` + `RxWebKit` to me. Anyway I get all the emails.

**Comment by lordzsolt** [comment]
[RxCoreData](https://github.com/RxSwiftCommunity/RxCoreData) is not in the list.

**Comment by jdisho** [comment]
I will take `RxKeyboard` and `RxAlamofire`.  @freak4pc

**Comment by Bruce-pac** [comment]
RxIGListKit <https://github.com/RxSwiftCommunity/RxIGListKit> is not in the list.

**Comment by freak4pc** [comment]
@jdisho Happy to put you on RxKeyboard. Let me know if you need any help getting that project up to speed with resolving issues/PRs/cutting releases etc.

RxAlamofire is a really large and important project that needs a lot to love, some sort of a roadmap and a more involved management, so before putting anyone on that I'd want to have a chat to make sure the person understands that project would require closer care :)

**Comment by freak4pc** [comment]
Add both, thanks @lordzsolt, @Bruce-pac. 

@Bruce-pac just wanted to make sure you're not still maintaining RxIGListKit? It not have gotten updates for 5 months. Just making sure :)

**Comment by rynecheow** [comment]
Am interested to look at maintaining RxAlamofire too - if we are open to a co-ownership setup.

**Comment by freak4pc** [comment]
@rynecheow was actually the active maintainer of RxAlamofire a few months ago so I’d feel comfortable with that. WDYT @jdisho

**Comment by jdisho** [comment]
Since there was no owner I volunteered to take it, but sure, makes more sense for @rynecheow 👍

**Comment by jdisho** [comment]
If no active maintainer comes for `RxAnimated` or `RxReachability`, feel free to put me in one of these. @freak4pc

**Comment by fassko** [comment]
I can take `RxAnimated` (together with @jdisho ) and `RxStarscream` @freak4pc 

And `NSObject-Rx` can be on me, although it is not mentioned here.

Have been contributed before on those ones.

**Comment by mRs-** [comment]
There is also `RxFileMonitor`. Don't know how the current situation is there.

**Comment by M0rtyMerr** [comment]
I can take `RxRealm`, used it a lot

**Comment by freak4pc** [comment]
@MortyMerr - RxRealm would be lucky to have you as a maintainer, writing you down there :) 

@jdisho - Putting you on RxKeyboard and RxAnimated, seems like two relatively small but meaningful projects you can create a lot of impact on. 

@fassko - Putting you on RxAnimated (Being secondary owner with Joan) and RxStarscream. IRT to NSObject-Rx, no problems. I didn't write it down because it's very minor so I didn't really think it requires maintenance. 

@rynecheow I'd love you to take over RxAlamofire if time allows, it needs some work - making sure everything is up to date... perhaps some tests, CI, updated docs. WDYT? 

Is there anything you need from my side to get started? I believe you are all members.

The first time you want to cut a release, ping me and I'll make sure you have everything set up so you can do it by yourself :)

**Comment by rynecheow** [comment]
@freak4pc Sounds good! Will probably start to take a look tomorrow and see what to cleanup for a start.

**Comment by freak4pc** [comment]
Amazing! Thanks everyone for coming through together for this community, you are the best! 💯

**Comment by freak4pc** [comment]
PS No need to enforce a specific CI, but I've been using GitHub Actions in a few projects lately and it's pretty awesome :)

**Comment by ashfurrow** [comment]
Sounds like a great plan, thank you @freak4pc 🙇 I have also found myself managing the release process for repos that I haven't contributed to in a few years so handing these off formally to people who are more actively involved sounds like a win/win.

@bobgodwinx I'm happy to hear you're keen on taking over the Action repo – I think you'll do an amazing job 💪

**Comment by Bruce-pac** [comment]
I am still maintaining RxIGListKit.

**Comment by freak4pc** [comment]
@Bruce-pac Perfect, writing you down. 

Please welcome @engali94 who'll be taking over RxFirebase 🔥

**Comment by twittemb** [comment]
Hi @freak4pc 

FYI, RxReduce is not under active development. I will archive it.

**Comment by freak4pc** [comment]
That's unfortunate, but thanks for letting me know @twittemb

**Comment by mosamer** [comment]
Hi @freak4pc I believe I'll also archive RxTestExt in favor of RxNimble.

**Comment by mRs-** [comment]
@fabianmuecke and I will take care of RxWebKit

**Comment by freak4pc** [comment]
Amazing, thank you @mRs- !

**Comment by o-nnerb** [comment]
Hi @freak4pc I can help with `RxReachability`

**Comment by minuscorp** [comment]
Hi! As I commented on Twitter, I'm available to maintain whichever package needs the most. Although RxAlamofire is my best preference, I can fit in any of them 🥰

**Comment by freak4pc** [comment]
@brennobemoura Help means taking ownership or regular contribution ? Happy for either :)

**Comment by freak4pc** [comment]
> Hi! As I commented on Twitter, I'm available to maintain whichever package needs the most. Although RxAlamofire is my best preference, I can fit in any of them 🥰

Hey, thank you so much! RxAlamofire already has two owners so it wouldn’t be as beneficial adding more there. RxMKMapView, RxCoreData and RxCoreMotion seem to be the ones left.

**Comment by o-nnerb** [comment]
@freak4pc I can help by being ownership of `RxReachability`

**Comment by freak4pc** [comment]
Ok, adding you @brennobemoura - note it should probably be upgraded at some point to conditionally use the new Network.framework stuff

**Comment by freak4pc** [comment]
Also, please welcome @rizwan95 who'll take over RxCoreMotion

**Comment by minuscorp** [comment]
I can help with RxMKMapView if needed

**Comment by freak4pc** [comment]
@minuscorp oh wow I'm sorry, I didn't see your comment. I'd be happy for you to take over RxMKMapView. Are you still interested? 

Also, the last open repo is RxCoreData if anyone's interested in that.

**Comment by minuscorp** [comment]
@freak4pc I'm not the CoreData type-of user haha 😅 so for now I can take RxMKMapView, will see forward if I can handle more libs

**Comment by freak4pc** [comment]
Sounds great, sorry for the delay in response here. 
Feel free to start working on RxMKMapView @minuscorp, let me know if you need any help. 
When you need access to upload a CocoaPod, let me know and I'll add you.

Cheerrs!

**Comment by o-nnerb** [comment]
@freak4pc I'm currently switching to Combine, so I would like to recommend @JoeMatt to continue updating RxReachability

**Comment by JoeMatt** [comment]
Yeah I could do that for now. It’s a pleasure. Someone just poke me if I’m snoozing on anything. 😁

**Comment by freak4pc** [comment]
Okie dokie :) Welcome, @JoeMatt !

**Comment by freak4pc** [comment]
FYI We're looking for a new maintainer/owner for RxRealm. Should be very little work but let me know if you're interested!

**Comment by rynecheow** [comment]
I don't mind to take this up too @freak4pc :)

**Comment by jegnux** [comment]
@freak4pc btw, if there's no current officiel maintainer, I'd like to take RxOptional

**Comment by freak4pc** [comment]
I got the imperssion @ashfurrow still sort of looks after it but don't have an issue with adding you as well.

@rynecheow Happy to! I'm sure you'll do great work. Adding you :)

**Comment by ashfurrow** [comment]
Yeah it originally came out of an app I built but I haven't been actively maintaining it for a while. Help would be appreciated @jegnux 🙇

**Comment by Econa77** [comment]
@freak4pc I can help with `RxWebKit`. Currently, RxSwift6 support is not completed.
I can also help `Action` and `RxAnimated` as well.

**Comment by mRs-** [comment]
RxWebKit is generally on our timeline, but we are currently lacking time for this (normal work is blocking us of this topic). But I try to get it this week done
